design/open/lakefs-on-kv-testing-plan.md

```diff
@@ -0,0 +1,128 @@
+# lakeFS on KV - Testing Plan
+
+## Global Testing
+
+### Migration
+* Global tests to verify and benchmark the migration process
+* Need to verify that the system behavior is not affected by the move to the KV DB
+* [Question] - Do we need to dispose of these once migration to KV is globally completed?
+
+#### Operational Testing
+* Set of tests that will run DB migration as part of the test
+* Currently, system tests assume lakeFS is up and running. Need to add ability to perform migration in the middle of the test (stop lakeFS -> run&verify migration -> start lakeFS ?)
+* Maybe something that can be 
+
+#### Data Level Testing
+* Dump (all) data before and after migration and compare it
+* Verifies data is preserved through migration, and is readable by the KV Store
+* Need to write appropriate dumpers. Can be done per package, as we go
+* Can be done both on DB package level, to verify the package itself, and on the using packages level, to verify data usage is not affected
+* [Question] - How do we define a covering data set, to test? How do we create
```

Comment:
This covers migration again I think.

Reply:
Correct. This is a section under the **Migration** headline. There are 2 sections: (1) Operational Tests - running the migration as part of a work flow, to verify the migration completes successfully and (probably) that  the system is stil operable, and (2) Data Level Tests - comparison of the data before and after the migration, for equality.
As I'm writing these, it seems that the first section covers the second too. Yet, I believe it will make it easier to test the migration code (at least for debugging purposes) an I think it should stay, at lease as a "Nice to Have".
Will add a note